### DISCUSSION TITLE
Bad Termination (segmentation fault) when splitting the mesh

### DISCUSSION THREAD

**Discussion by morsi95** [idaholab/moose#23713] [Q&A Meshing] [answered]
Dear all,

I receive a segmentation fault error after running the following command on my input file. The mesh block in the input file looks like this:
```
[Mesh]
  [./file_mesh]
    type = FileMeshGenerator
    file = mesh_test.e
  []
[]
```
The command is:
mpiexec moose-app-opt  --n-threads=7--split-mesh 7 -i PT_th1.i

Firstly, I received this  error:
Output mesh file name must be specified (with --split-file) when splitting non-file-based meshes

Then when I added the --split-file flag I received the BAD TERMINATION (segmentation fault) error when it started to write the splitted files.

Could someone help?

**Comment by GiudGiud** [comment]
Does it work with 
```
mpiexec -n 7 moose-app-opt  --split-mesh 7 -i PT_th1.i --split-file my_split
```

I m not sure the splitting mesh supports threading

Also,
does this installation of MOOSE pass the test suite (run_tests in moose/test)?

**Comment by morsi95** [comment]
I tried --split-file but still didn't work.

Here is the error log that I receive when I run:
./run_tests -j --failed-tests
[failed_tests.txt](https://github.com/idaholab/moose/files/10989625/failed_tests.txt)

**Reply by GiudGiud** [reply]
These diffs are problematic but they also all pertain to some of the more numerically challenging problems so we can let it slide for now. 

I need an error message. Or your mesh so I can try to split it locally

**Reply by morsi95** [reply]
I tried the command on a different moose installation on my local machine but I received the following error:
```pre
Setting Up............
  Setting Up Undisplaced Mesh
    Preparing Mesh
      Updating Mesh...                                                                   [ 20.79 s] [    0 MB]
    Finished Preparing Mesh                                                              [ 21.37 s] [    0 MB]
  Finished Setting Up Undisplaced Mesh                                                   [ 21.37 s] [    0 MB]
Splitting 42 ways...
Still Setting Up...............    - writing 42 files per process...
..
===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 8619 RUNNING AT mo
=   EXIT CODE: 139
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```

**Reply by GiudGiud** [reply]
How big is the file? How many elements?
Are you running out of memory by any chance?

**Reply by GiudGiud** [reply]
I m still unsure why you are using split mesh btw.
What is the problem with replicated meshes?

**Reply by morsi95** [reply]
My mesh is 9 million elements so in order to optimize the running time on a hpc cluster I need to be able to use distributed mesh. But with --distributed-mesh it looks like the RAM is overloaded so a pre splitting is probably needed.

**Reply by GiudGiud** [reply, marked as answer]
Ok that's the right reason for this. 
What is the split going to be?
I see 42 splits per process, times 7?
